enhancements/insights/up-to-date-gathering.md

```diff
@@ -0,0 +1,250 @@
+
+## Motivation
+
+The motivation is to be able to react faster on requirements for additional
+Insights Operator data requests. The benefit for customers would be faster
+identification of clusters based on newly discovered issues, identified by
+particular cluster configuration.
+The other potential slowdown might be caused by fact that administrators need
+to upgrade their clusters to deliver actual updates (either automatically or manually).
+
+Example of situation when the process is too slow:
+Two weeks ago we discovered an issue that for larger amount of generated alerts, IO Archives
+becomes too large (#148, CCXDEV-2673). We decided to fix it in all available versions (4.6, - 4.3)
+We spent 3 days on development and review until it was merged in Master/4.6
+3 Days later we created PR for 4.5, it was merged 6 days later.
+6 days after we creted PR for 4.4 it is still not merged (now e2e tests are failing)
+We still need to merge it to 4.3.
+Since the beginning it is now 24 days.
+
+When CCX needs release faster: 
+- To identify affected customers (suggest Workaround), 
+- To find real feature usage, especially in in historical releases (because all that waiting on merge)
+- Fix IO bug
+
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Release process change. On a fast-track release, publish Insights Operator fast-track changes during build to new quay tag for example 4.5-fast-track.
+The rest of release process will behave like today, release to 4.x.
+
+There will be two containers in Insights Operator Pod (*main* container with driver and *gatherer* container with Insights Operator).
+* Run the gathering logic in a separate container: a *gatherer* 
+* Control the lifecycle of the gatherer from *main* insights-operator container (different executable as part of the same image): this includes using newer versions
+of the image when available, as well as fallback mechanism to older version on failures.
+
+### Non-Goals
+
+
+## Proposal
+
+Separate The Insights Operator pod in two containers the Operator *main* and 
+Insighs Operator *gatherer*.
+
+The *main* will be similar to today's operator. It would like a proper operator 
+drive *gatherer*, but keep all the non-gathering logic, like uploading, status updates etc in it.
+It will be periodically checking for new versions in a 4.x-fast-track image on quay.io and eventually update the
+Insights Operator Gatherer container to latest 4.x-fast-track image url.
+It will not contain the Gathering methods and it will use Rest to reach *gatherer*, which will be 
+responsible for collection and saving data to storagePath (/var/lib/insights-operator) and this volume will be 
+shared among two containers.
+The *main* will instead of local Gather call remote Gather as Api, but will keep all the status handling and uploading
+in *main*.
+
+The Insights Operator *gatherer* will become just rest server on operators framework exposing Gather endpoint.
```

Comment:
I don't understand this.  The gatherer isn't a server, it's a client (to other things) and then writes things to disk.  The uploader takes things from disk and moves them to insights.

I do not see a justification for a server endpoint for the gatherer or for the uploader.